fbw-common/src/wasm/extra-backend/MsfsHandler/DataTypes/ClientEvent.h

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) 2022 FlyByWire Simulations
+// SPDX-License-Identifier: GPL-3.0
+
+#ifndef FLYBYWIRE_AIRCRAFT_CLIENTEVENT_H
+#define FLYBYWIRE_AIRCRAFT_CLIENTEVENT_H
+
+#include <functional>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <MSFS/Legacy/gauges.h>
+#include <SimConnect.h>
+
+#include "IDGenerator.h"
+
+class DataManager;
+
+// Used for callback registration to allow removal of callbacks
+typedef uint64_t CallbackID;
+
+/**
+ * Defines a callback function for an event
+ * @param number of parameters to use - TODO: maybe remove this
+ * @param parameters 0-4 to pass to the callback function
+ */
+typedef std::function<void(int number, DWORD param0, DWORD param1, DWORD param2, DWORD param3, DWORD param4)> EventCallbackFunction;
+
+/**
+ * @brief The ClientEvent class represents a client event which can be used to create a custom event,
+ * or be mapped to a sim event, or be mapped to a system event.<p/>
+ *
+ * A ClientEvent has a unique id and a name. The name can be used to map the event to a sim event or
+ * to create a custom event.<br/>
+ * Custom events must have a name that contains a period (e.g. "Custom.Event") to the sim recognizes
+ * it as a custom event.<br/>
+ * To map to sim events the name must be identical to the name of the sim event otherwise there will be
+ * a SimConnect exception that the event is unknown.<br/>
+ * If the ClientEvent is intended to be used as a system event then it must be constructed with the
+ * registerToSim parameter set to false. This will prevent the event from being registered to the sim.
+ * See constructor documentation for more details.
+ */
+class ClientEvent {
+ private:
+  // allow DataManager to access the private constructor
+  friend DataManager;
+
+  // allow the streaming operator to access the private members
+  friend std::ostream& operator<<(std::ostream& os, const ClientEvent& event);
+
+  // Simconnect handle
+  HANDLE hSimConnect;
+
+  // the id of the client event
+  SIMCONNECT_CLIENT_EVENT_ID clientEventId;
+
+  // the name of the client event - for custom events this should contain a period
+  const std::string clientEventName{};
```

Comment:
why is it initialized is the question :D - removed the {}
otherwise I think it does not make sense to reuse an event instance with a different name (might even be problematic if the ID is already used)